homeassistant/components/unifi/services.py

```diff
@@ -0,0 +1,69 @@
+"""UniFi services."""
+
+from .const import DOMAIN as UNIFI_DOMAIN
+
+UNIFI_SERVICES = "unifi_services"
+
+SERVICE_REMOVE_CLIENTS = "remove_clients"
+
+
+async def async_setup_services(hass) -> None:
+    """Set up services for UniFi integration."""
+    if hass.data.get(UNIFI_SERVICES, False):
+        return
+
+    hass.data[UNIFI_SERVICES] = True
```

Comment:
Shouldn't we keep this within the domain of this integration?
`hass.data[DOMAIN][SERVICES]`

Reply:
Do we really need to check again registering this double? Like would it matter?

Reply:
This is done like in the deConz integration. Both are quite similar in regards to supporting multiple instances and having services. 
Right now the integrations reside directly under hass data unifi. I would need to add another layer for every place retrieving an instance of the integration if we want to move services to hass data unifi services.

Reply:
If there is no control the first integration to be removed would remove the services as well? Maybe I don't understand your second question?

Reply:
Aah ok, I should just move the verification to the same level when setting up services you mean?

Reply:
That would have the same effect. Additionally, I don't think (not 100% sure, @MartinHjelmare can probably confirm), registering a service twice isn't a problem / doesn't have side effects. So I think we don't need the logic that prevents them from registering twice.

Reply:
It will send out multiple events https://github.com/home-assistant/core/blob/55328d2c6fb66deb876e5c4647f9446ad279d350/homeassistant/core.py#L1351 and that's about it, I'm just trying to keep stuff tidy :)

Reply:
Hmmm right :)
I guess we send out a duplicate event quite a bit in general (in case of multiple config entries).

I wonder if that is something we should solve on a different level 🤔 

Anyways, it's fine 👍  I think the `UNIFI_SERVICES` can be refactored out while still achieving the same and some docs and it should be good to go.

Reply:
Can we merge this as is and I will do a follow up PR to clean this up tonight? Would be nice to not miss the beta

Reply:
Added documentation PR

Reply:
I think that is a good deal 😉 👍

Reply:
Resolved in #56834